net/graphsync_fetcher.go

```diff
@@ -185,3 +240,56 @@ func (gsf *GraphSyncFetcher) loadTipsetFromCids(ctx context.Context, cids []cid.
 
 	return types.NewTipSet(validatedBlocks...)
 }
+
+func (gsf *GraphSyncFetcher) missingCids(cids []cid.Cid) ([]cid.Cid, error) {
+	missing := make([]cid.Cid, 0, len(cids))
+	for _, cid := range cids {
+		hasBlock, err := gsf.store.Has(cid)
+		if err != nil {
+			return nil, err
+		}
+		if !hasBlock {
+			missing = append(missing, cid)
+		}
+	}
+	return missing, nil
+}
+
+func (gsf *GraphSyncFetcher) areCidsMissing(tsKey types.TipSetKey) (bool, error) {
+	missing, err := gsf.missingCids(tsKey.ToSlice())
+	if err != nil {
+		return false, err
+	}
+	if len(missing) != 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+type requestPeerFinder struct {
```

Comment:
Please document this abstraction, especially its stateful nature and intended use pattern.